apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/FormSummary/index.tsx

```diff
@@ -43,11 +47,14 @@ const SwapFormSummary = ({ swapTransaction, provider }: SwapFormSummaryProps) =>
   } = swapTransaction.swap.from;
   const { currency: toCurrency, account: toAccount } = swapTransaction.swap.to;
   const ratesState = swapTransaction.swap.rates;
-  const hasRates = ratesState?.value?.length && ratesState?.value?.length > 0;
+  const hasRates = ratesState?.value?.length && ratesState?.value?.length > 0; // TODO why do we need this and the below line? They appear to be the same thing
```

Comment:
That's quite odd indeed.

We should be able to do without one of them.

As I understand it, the `hasRates` is a shorthand for the default value of the `hasFetchedRates` state value and this `hasFetchedRates` state value is updated every time `hasRates` changes (through the `useEffect`).

Two possibilities:
- Remove the `hasFetchedRates` line and `useEffect` update logic entirely and only use `hasRates`.
This should work fine if `ratesState?.value?.length` is not unassigned or updated to 0 between renders. Because with today's logic, once `hasRates` is `true` once, `hasFetchedRates` is set to `true` and not updated anymore.
Because `setHasFetchedRates(v => (!v ? hasRates : v))` updates the `hasFetchedRates` value (to `hasRates`) only if the previous value of `hasFetchedRates` was falsy. If the previous value was truthy, it reassigns it to the same value (therefore not updating the actual value).

- Keep the `hasRates` and `hasFetchedRates` logic (and maybe rename `hasFetchedRates` to `hasFetchedRatesOnce` to be more explicit) and only use `hasFetchedRates` in the returned component. 
In the returned component, both `hasFetchedRates` and `hasRates` are used. This seems odd and, based on current logic, only `hasFetchedRates` should be used (since it is derived from `hasRates`). Also, `hasFetchedRates` and `hasRates` *may* have different values at any point in time (cf. explanation above)

Reply:
It sounds like keeping the useEffect might be safer but keeping it is potentially redundant. I will try the first suggestion and test it to see if there are any obvious issues

Reply:
I've tested both keeping just `hasRates` and keeping the useEffect that responds to the rates changes and both seem to be the exact same behaviour. It's just controlling the account selection and fee selection buttons and the main swap form is the main control whether it displays or not, so I think having the simplest solution (`hasRates`) is fine